File: front/src/app/model/negociacao.ts
import { SindicatoPatronal } from './sindicato-patronal';
import { SindicatoLaboral, Mes } from './sindicato-laboral';
import { Empresa } from './empresa';

export enum StatusNegociacao {
    NaoIniciada = 'Não Iniciada',
    EmNegociacao = 'Em Negociação',
    Fechada = 'Fechada',
    Dissidio = 'Dissídio'
}

export class Negociacao {
    id: number;
    ano: number;
    empresaId: number;
    sindicatoLaboralId: number;
    sindicatoPatronalId: number;
    orcadoId: number;
    negociadoId: number;
    qtdaTrabalhadores: number;
    massaSalarial: number;
    taxaLaboral: number;
    taxaPatronal: number;
    statusACT_CCT: StatusNegociacao;
    statusPLR: StatusNegociacao;
    qtdaRodadas: number;
    plr1Sem: number;
    plr2Sem: number;
    custosViagens: number;

    empresa: Empresa;
    sindicatoLaboral: SindicatoLaboral;
    sindicatoPatronal: SindicatoPatronal;
    orcado: Reajuste;
    negociado: Reajuste;

    rodadasNegociacoes: RodadaNegociacao[];
}

export class RodadaNegociacao {
    id: number;
    negociacaoId: number;
    numero: number;
    data: Date;
    resumo: string;
    custosViagens: number;
    negociacao: Negociacao;
}

export class Reajuste {
    id: number;
    negociacaoId: number;
    salario: number;
    piso: number;
    auxCreche: number;
    vaVr: number;
    vaVrFerias: number;
    descontoVt: number;

    parcelas: ParcelaReajuste[];
}

export class ParcelaReajuste {
    id: number;
    reajusteId: number;
    mes: Mes;
    valor: number;
}

export class Concorrente {
    id: number;
    nome: string;
    negociacaoId: number;
    reajusteId: number;

    reajuste: Reajuste;
}

File: GestaoSindicatos/Model/Negociacao.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace GestaoSindicatos.Model
{

    public enum StatusNegociacao
    {
        NaoIniciada,
        EmNegociacao,
        Fechada,
        Dissidio

    }
    public class Negociacao
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int Ano { get; set; }
        public int? EmpresaId { get; set; }
        public int? SindicatoLaboralId { get; set; }
        public int? SindicatoPatronalId { get; set; }
        [Required]
        public int QtdaTrabalhadores { get; set; }
        [Required]
        public double MassaSalarial { get; set; }
        public int? QtdaRodadas { get; set; }
        public float? TaxaLaboral { get; set; }
        public float? TaxaPatronal { get; set; }
        public int? OrcadoId { get; set; }
        public int? NegociadoId { get; set; }
        public float? Plr1Sem { get; set; }
        public float? Plr2Sem { get; set; }
        [Required]
        public StatusNegociacao StatusACT_CCT { get; set; }
        [Required]
        public StatusNegociacao StatusPLR { get; set; }


        public virtual Empresa Empresa { get; set; }
        public virtual SindicatoLaboral SindicatoLaboral { get; set; }
        public virtual SindicatoPatronal SindicatoPatronal { get; set; }
        [ForeignKey("OrcadoId")]
        public virtual Reajuste Orcado { get; set; }
        [ForeignKey("NegociadoId")]
        public virtual Reajuste Negociado { get; set; }

        public virtual ICollection<Concorrente> Concorrentes { get; set; }
        [JsonIgnore]
        public virtual ICollection<RodadaNegociacao> RodadasNegociacoes { get; set; }

        [NotMapped]
        public float CustosViagens
        {
            get
            {
                if (RodadasNegociacoes == null || RodadasNegociacoes.Count <= 0)
                    return 0F;

                return RodadasNegociacoes.Sum(r => r.CustosViagens ?? 0);
            }
        }
    }
}
